src/_gettsim/transfers/wohngeld.py

```diff
@@ -66,29 +66,134 @@ def wohngeld_m_wthh(
+
+    Returns
+    -------
+
+    """
+    if wohngeld_anspruchsbedingungen_erfüllt_wthh:
+        out = wohngeld_basisbetrag_m_wthh
+    else:
+        out = 0.0
+
+    return out
+
+
+def wohngeld_anspruchshöhe_bg(
+    wohngeld_basisbetrag_m_bg: float,
+    wohngeld_anspruchsbedingungen_erfüllt_bg: bool,
+) -> float:
+    """Housing benefit after wealth and income check.
+
+    This target is used for the priority check calculation against ALG2 and
+    Kinderzuschlag.
+
+    Parameters
+    ----------
+    wohngeld_basisbetrag_m_bg
+        See :func:`wohngeld_basisbetrag_m_bg`.
+    wohngeld_anspruchsbedingungen_erfüllt_bg
+        See :func:`wohngeld_anspruchsbedingungen_erfüllt_bg`.
+
+    Returns
+    -------
+
+    """
+    if wohngeld_anspruchsbedingungen_erfüllt_bg:
+        out = wohngeld_basisbetrag_m_bg
+    else:
+        out = 0.0
+
+    return out
+
+
+def wohngeld_anspruchsbedingungen_erfüllt_wthh(
+    wohngeld_mindesteinkommen_erreicht_wthh: bool,
+    wohngeld_vermögensgrenze_unterschritten_wthh: bool,
+) -> bool:
+    """Check whether the household meets the conditions for Wohngeld.
+
+    This target is used to calculate the actual Wohngeld of the Bedarfsgemeinschaften
+    that passed the priority check against ALG2 and Kinderzuschlag.
+
+    Parameters
+    ----------
+    wohngeld_mindesteinkommen_erreicht_wthh
+        See :func:`wohngeld_mindesteinkommen_erreicht_wthh`.
+    wohngeld_vermögensgrenze_unterschritten_wthh
+        See :func:`wohngeld_vermögensgrenze_unterschritten_wthh`.
+
+    Returns
+    -------
+
+    """
+    return (
+        wohngeld_vermögensgrenze_unterschritten_wthh
+        and wohngeld_mindesteinkommen_erreicht_wthh
+    )
+
+
+def wohngeld_anspruchsbedingungen_erfüllt_bg(
+    wohngeld_mindesteinkommen_erreicht_bg: bool,
+    wohngeld_vermögensgrenze_unterschritten_bg: bool,
+) -> bool:
+    """Check whether the household meets the conditions for Wohngeld.
+
+    This target is used for the priority check calculation against ALG2 and
+    Kinderzuschlag.
+
+    Parameters
+    ----------
+    wohngeld_mindesteinkommen_erreicht_bg
+        See :func:`wohngeld_mindesteinkommen_erreicht_bg`.
+    wohngeld_vermögensgrenze_unterschritten_bg
+        See :func:`wohngeld_vermögensgrenze_unterschritten_bg`.
+
+    Returns
+    -------
+
+    """
+    return (
+        wohngeld_mindesteinkommen_erreicht_bg
+        and wohngeld_vermögensgrenze_unterschritten_bg
+    )
+
+
 def wohngeld_abzüge_st_sozialv_m(
-    eink_st_y_sn: float,
+    lohnst_m: float,
     ges_rentenv_beitr_arbeitnehmer_m: float,
     ges_krankenv_beitr_arbeitnehmer_m: float,
     kind: bool,
     wohngeld_params: dict,
 ) -> float:
     """Calculate housing benefit subtractions on the individual level.
 
-    Note that eink_st_y_sn is used as an approximation for taxes on income (as mentioned
-    in § 16 WoGG Satz 1 Nr. 1).
-
     Parameters
     ----------
-    eink_st_y_sn
-        See :func:`eink_st_y_sn`.
+    lohnst_m
```

Comment:
Oh, I am sorry if I suggested this! This will be wrong for the self-employed, I am afraid. Not sure how it is actually determined for those, but I'd rather leave it as it was.

Reply:
Hm then I don't understand your comment regarding switching to Lohnsteuer in `arbeitsl_geld_2_nettoeink_ohne_freibetrag_m`. There, the income for self-employed would be treated wrong as well (because we now subtract Lohnsteuer instead of `eink_st_y_sn`). Can you explain what you meant back then?